kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/subject/DefaultTransportSubjectBuilderServiceTest.java

```diff
@@ -0,0 +1,839 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletionException;
+
+import javax.security.auth.x500.X500Principal;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+
+import io.kroxylicious.proxy.authentication.Subject;
+import io.kroxylicious.proxy.authentication.TransportSubjectBuilder;
+import io.kroxylicious.proxy.authentication.User;
+import io.kroxylicious.proxy.tls.ClientTlsContext;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class DefaultTransportSubjectBuilderServiceTest {
+
+    private static YAMLMapper mapper = new YAMLMapper();
+
+    private static void tls(DefaultTransportSubjectBuilderService.Config builderConfig,
+                            X509Certificate cert,
+                            @Nullable Subject expectedSubject,
+                            @Nullable Exception expectedException) {
+        if ((expectedSubject != null) == (expectedException != null)) {
+            throw new IllegalArgumentException("These arguments are mutually exclusive.");
+        }
+        var service = new DefaultTransportSubjectBuilderService();
+        service.initialize(builderConfig);
+        TransportSubjectBuilder builder = service.build();
+        var subject = builder.buildTransportSubject(new TransportSubjectBuilder.Context() {
+            @Override
+            public Optional<ClientTlsContext> clientTlsContext() {
+                return Optional.of(new ClientTlsContext() {
+                    @Override
+                    public X509Certificate proxyServerCertificate() {
+                        return null;
+                    }
+
+                    @Override
+                    public Optional<X509Certificate> clientCertificate() {
+                        return Optional.of(cert);
+                    }
+                });
+            }
+        });
+
+        Assertions.assertThat(subject).isCompleted();
+        var fut = subject.toCompletableFuture();
+        if (expectedSubject != null) {
+            Assertions.assertThat(fut.join())
+                    // .extracting(Subject::principals, InstanceOfAssertFactories.set(Principal.class))
+                    // .singleElement()
+                    // .isInstanceOf(User.class)
+                    // .extracting(Principal::name, InstanceOfAssertFactories.STRING)
+                    .isEqualTo(expectedSubject);
+        }
+        else {
+            Assertions.assertThat(fut)
+                    .isCompletedExceptionally();
+            assertThatThrownBy(() -> fut.join()).isExactlyInstanceOf(CompletionException.class)
```

Comment:
I think there is a better fix than the one sonar is suggesting but In testing I don't think this assertion is being used.